Max-Movies-In-Years-2.js

```diff
@@ -0,0 +1,67 @@
+function displayMap(map) {
+    map.forEach((value,key)=>console.log(key+":"+value))
+    }
+
+    //Response consist of obj ->year [start,end]
+    const  arr = [
+    {
+    name : "kratika",
+    year :[2008,2010]
+    },
+    {
+    name : "kratika2",
+    year :[2005,2012]
+    },
+    {
+    name : "kratika3",
+    year :[2013,2014]
+    },
+    {
+    name : "kratika4",
+    year:[2007,2011]
+    }
+    ]
+
+    function getMaximumYears2(arr) {
+
+        // get the years and sort to get lowest and highest year 
+        const years = arr.reduce((acc,element)=> {
+            return acc.concat(element.year)
+            },[]).sort((a,b)=>a-b)
+       
+       const minYear =years[0];
+       const maxYear = years[years.length-1];
+       // map year to movie 
+       let yearMap = new Map();
+       for (let i = minYear ; i<= maxYear ; ++i) {
+           yearMap.set(i,0);
+       }
+       
+       arr.forEach((element)=>{
+           for (let i = element.year[0] ; i<=element.year[1];++i) {
+                yearMap.set(i,(yearMap.get(i)+1));
+       }
```

Comment:
You can check if the map already has it and then add, or set as 1

Less code = Fewer mistakes + More confidence